CMakeLists.txt

```diff
@@ -1431,6 +1432,26 @@ add_subdirectory("examples")
 add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND} "--output-on-failure" -j ${CORECOUNT} ${COMMAND_USES_TERMINAL})
 add_custom_target(check_tier1 COMMAND ${CMAKE_CTEST_COMMAND} "--output-on-failure" -L "'internal|amdsdk_30|piglit|PyOpenCL|conformance_suite_micro'" -j ${CORECOUNT} ${COMMAND_USES_TERMINAL})
 
+set(CPACK_PACKAGE_NAME pocl)
+set(CPACK_PACKAGE_VENDOR pocl)
+set(CPACK_PACKAGE_DESCRIPTION_FILE "${CMAKE_SOURCE_DIR}/CPack.pocl.description.txt")
+set(CPACK_PACKAGE_HOMEPAGE_URL "https://github.com/pocl/pocl")
+set(CPACK_PACKAGE_CONTACT "https://github.com/pocl/pocl")
+set(CPACK_PACKAGE_CHECKSUM "SHA512")
+set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_SOURCE_DIR}/COPYING")
+set(CPACK_GENERATOR "DEB")
+
+string(REPLACE "." ";" LLVM_VERSION_PARSED "${LLVM_VERSION}")
+list(GET LLVM_VERSION_PARSED 0 LLVM_VERSION_MAJOR)
+list(GET LLVM_VERSION_PARSED 1 LLVM_VERSION_MINOR)
+
+set(CPACK_DEBIAN_PACKAGE_DEPENDS "clang-${LLVM_VERSION_MAJOR} (>= ${LLVM_VERSION_MAJOR}),libgcc1,libhwloc5 (>= ${Hwloc_VERSION}),libllvm${LLVM_VERSION_MAJOR} (>= ${LLVM_VERSION_MAJOR}),libc6")
```

Comment:
Hwloc is optional, and also LLVM is optional, so the dependencies here should be ifdef-ed on ENABLE_HWLOC and OCS_AVAILABLE cmake variables.

Reply:
Thanks. Gonna fix.

Reply:
Clang/LLVM? I think the main problem there would be possibly mismatching versions between the builtin IR lib and the Clang/LLVM it happens to find at runtime.

Reply:
No, hwloc and ocs.